state/applicationoffers_test.go

```diff
@@ -566,6 +653,27 @@ func (s *applicationOffersSuite) TestRemoveOffersWithConnectionsForce(c *gc.C) {
 	s.assertInScope(c, mysqlru, true)
 }
 
+func (s *applicationOffersSuite) TestRemovingApplicationFailsRace(c *gc.C) {
+	s.createDefaultOffer(c)
+	wp := s.AddTestingApplication(c, "local-wordpress", s.AddTestingCharm(c, "wordpress"))
+
+	bumpRelCount := func() {
+		eps, err := s.State.InferEndpoints(wp.Name(), s.mysql.Name())
+		_, err = s.State.AddRelation(eps...)
+		c.Assert(err, jc.ErrorIsNil)
+	}
+
+	bumpRevno := func() {
+		_, err := s.mysql.AddUnit(state.AddUnitParams{})
+		c.Assert(err, jc.ErrorIsNil)
+	}
+
+	defer state.SetBeforeHooks(c, s.State, bumpRelCount, bumpRevno, bumpRevno).Check()
+
+	err := s.mysql.Destroy()
+	c.Assert(err, gc.ErrorMatches, "cannot destroy application.*")
```

Comment:
We want to assert that the error also satisfies IsNotSupported
(we need the typed error to pass to the CLI)

Reply:
These errors don't trigger `txn.ErrAborted`. Do you want to cover the blanket case?

func (op *DestroyApplicationOperation) Done(err error) error {
	if err == txn.ErrAborted {
		connected, err2 := applicationHasConnectedOffers(op.app.st, op.app.Name())
		if err2 != nil {
			err = errors.Trace(err2)
		} else if connected {
			err = errors.NotSupportedf("newly related offer")
		}
	}


	err = errors.NotSupportedf("change to the application detected")
	return errors.Annotatef(err, "cannot destroy application %q", op.app)
}